lightning/src/ln/channel.rs

```diff
@@ -695,7 +699,8 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 			match &msg.shutdown_scriptpubkey {
 				&OptionalField::Present(ref script) => {
 					// Peer is signaling upfront_shutdown and has provided a non-accepted scriptpubkey format. We enforce it while receiving shutdown msg
-					if script.is_p2pkh() || script.is_p2sh() || script.is_v0_p2wsh() || script.is_v0_p2wpkh() {
+					let is_witness_program_allowed = their_features.supports_shutdown_anysegwit() && script.is_witness_program();
```

Comment:
Currently this PR checks holder node supports `shutdown_anysegwit` at `shutdown` message reception only.

It has the following pitfall. If our peer provide us a anysegwit scriptpubkey at channel opening through the `upfront_shutdown` feature we'll accept it but reject it later during the cooperative closure attempt, thus forcing us to fallback on a costly, privacy-damaging unilateral closure.

If we don't support `shutdown_anysegwit`, I think we should reject channel opening at all. It's up to our peer to read our public features correctly and propose us an upfront witness suiting us.

Reply:
Hmmm, so if I'm reading this right, you're suggesting that if they say they don't support shutdown anysegwit, but the script is witness, we fail right away?